### ISSUE TITLE
Questions?

### ISSUE THREAD

**Issue by jackokring** [expertsleepersltd/disting#3]
Very nice. But ...

I have code experience so, not that (done VST2 and VCVRack). General interest and coding possibilities.

1. 112 up to 127 seem to be favourite selects, why does the algorithm function pointers stop at 107 and not 111?
2. 64 to 87 enable SD card (ok), is the file system `fopen` etc., or some other way for getting files (haven't found the filesystem code ...)? And this range of algorithms is best kept for SD card use for consistency?
3. Inputs are a little scarce at 3, (XYZ), would a XYZ to AB->XY with a Z for a "pair mode" (delta coding?) and 2 distings work? I'm sure some 4 input specials might be good.
4. I might not develop, but making the reply general or on the Wiki would be good. (more of a data request).
5. A simplistic float -> (int, int) ADT with normalisation for output basic set of +* would assist many. Any plans to extend the "source"?

**Comment by expertsleepers** [comment]
1) That's most likely an oversight.
2) It relies on the Microchip libraries for accessing the SD. There's no reason to keep this particular range.
3) No idea what you mean, sorry.
4) OK.
5) No plans to extend this source code.

**Comment by jackokring** [comment]
3. Suppose 2 distings are available. XYZ can map to AB as some coded standard, and use just XY on the second disting. So XYZ and Z on the second become CV.

So given 96KHz sample rate, with AB channels, then effectively that's 4 44.1KHz channels and a control stream. So XYZ and sync.

**Comment by jackokring** [comment]
3. I've thought about it and a good implementation might be XY to A, and Z plus some parameter to B. If the mode itself can set up persistent flags (on/off) such that other modes on a second device have slightly lower audio quality multi CV input on X or Y from the `LFO` and low audio rate primary disting. I think checking a few flags with some extra input processing rather than loads of extra duplicated modes might be better. Not sure of the MIPS on it ...

**Comment by ibz** [comment]
Another question: I am looking at https://github.com/expertsleepersltd/disting/blob/master/hello_disting_mk4/main.c#L329C49-L329C49

Why are all algorithms `sumDifference`? Is it because the actual algorithm implementations are not open source and just the framework is, so the algorithms are built separately somehow? :)

**Comment by expertsleepers** [comment]
Yes. This isn't the actual disting source code, it's just an open source framework which is similar to the actual code.

**Comment by ibz** [comment]
Thanks for the quick, although disappointing, answer. But it's what I realized after lots of digging. Was hoping to see the actual algorithms. Cheers!
